File: leetcode/1480. Running Sum of 1d Array/s3.py
# OJ: https://leetcode.com/problems/running-sum-of-1d-array/
# Author: github.com/punkfulw
# Time: O(N)
# Space: O(1)
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        return accumulate(nums)

File: leetcode/300. Longest Increasing Subsequence/s3.cpp
// OJ: https://leetcode.com/problems/longest-increasing-subsequence/
// Author: github.com/punkfulw
// Time: O(NlogN)
// Space: O(N)
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> ans;
        for (auto i: nums){
            auto it = lower_bound(ans.begin(), ans.end(), i);
            if (it == ans.end())
                ans.push_back(i);
            else
                *it = i;
        }
        return ans.size();
    }
};
